force-app/main/default/classes/JobApplicationHelper.cls

```diff
@@ -121,4 +121,41 @@ public with sharing class JobApplicationHelper {
             tasksToInsert.add(TaskManager.prepareTask(application, taskType, priority, subject));
         }
     }
+
+    public static void setPrimaryContact(List<Job_Application__c> jobAppList) {
+        Set<Id> accountIds = new Set<Id>();
```

Comment:
what if we add a null check for both the primary contact OR company_name is null before the set and add a return statement like you did on line 135?

Reply:
I added that comment, but then remembered we talked about making the Company Name field required for a job app record. Maybe its not necessary since we know one of those will never be null